src/core/cgroup.c

```diff
@@ -1853,6 +1865,61 @@ static int unit_realize_cgroup_now_enable_breadth_first(Unit *u, ManagerState st
         return 0;
 }
 
+/* Controllers can only be disabled depth-first, from the leaves of the
+ * hierarchy upwards to the unit in question. */
+static int unit_realize_cgroup_now_disable_depth_first(Unit *u, ManagerState state) {
+        Iterator i;
+        Unit *m;
+        void *v;
+
+        assert(u);
+
+        if (u->type != UNIT_SLICE)
+                return 0;
+
+        HASHMAP_FOREACH_KEY(v, m, u->dependencies[UNIT_BEFORE], i) {
+                CGroupMask target_mask, enable_mask;
+
+                if (m == u)
+                        continue;
```

Comment:
Is this actually ever hit? I think we filter self from the "before" dependency list always.